pitop/processing/algorithms/line_detect.py

```diff
@@ -0,0 +1,116 @@
+from numpy import arctan, array, pi
+
+from pitop.processing.utils.vision_functions import (
+    colour_mask,
+    find_centroid,
+    find_largest_contour,
+    scale_frame,
+)
+from pitop.camera.pil_opencv_conversion import (
+    pil_to_opencv,
+    opencv_to_pil,
+)
+
+
+def import_opencv():
+    try:
+        import cv2
+        return cv2
+    except (ImportError, ModuleNotFoundError):
+        raise ModuleNotFoundError(
+            "OpenCV Python library is not installed. You can install it by running 'sudo apt install python3-opencv'.") from None
+
+
+def calculate_blue_limits():
+    # define range of blue color in HSV-> H: 0-179, S: 0-255, V: 0-255
+    # broken out like this for easy conversion into CV units (will move somewhere more logical in future)
+    hue_lower = 160
+    hue_upper = 280
+    sat_lower = 0.3
+    sat_upper = 1.0
+    val_lower = 0.5
+    val_upper = 1.0
+    cv_hue_lower = int(hue_lower / 2)
+    cv_hue_upper = int(hue_upper / 2)
+    cv_sat_lower = int(sat_lower * 255)
+    cv_sat_upper = int(sat_upper * 255)
+    cv_val_lower = int(val_lower * 255)
+    cv_val_upper = int(val_upper * 255)
+    lower_blue = array([cv_hue_lower, cv_sat_lower, cv_val_lower])
+    upper_blue = array([cv_hue_upper, cv_sat_upper, cv_val_upper])
+    return (lower_blue, upper_blue)
+
+
+def find_line(frame):
+    scale_factor = 0.5
+    cv_frame = pil_to_opencv(frame)
+
+    resized_frame = scale_frame(cv_frame, scale=scale_factor)
+    hsv_lower, hsv_upper = calculate_blue_limits()
+    image_mask = colour_mask(resized_frame, hsv_lower, hsv_upper)
+    line_contour = find_largest_contour(image_mask)
+
+    centroid = None
+    scaled_image_centroid = None
+    if line_contour is not None:
+        # find centroid of contour
+        scaled_image_centroid = find_centroid(line_contour)
+        centroid = centroid_reposition(scaled_image_centroid, 1, resized_frame)
+
+    robot_view_img = robot_view(resized_frame, image_mask, line_contour, scaled_image_centroid)
+
+    return (centroid, opencv_to_pil(robot_view_img))
```

Comment:
I think we should split this out, so that "find line" returns a line, and further processing derives subsequent useful output (such as 'robot view', which is really just a render of the line to an image).